docs/important/relevant_functionalities_de.md

```diff
@@ -0,0 +1,1250 @@
+
+# liest Konfigurationwert aus /config/<dogu>/my_key, gibt einen Defaultwert zurück, wenn dieser nicht gesetzt wurde
+$ doguctl config --default NOVALUE my_key2
+NOVALUE
+
+# liest verschlüsselten DB-Namen /config/<dogu>/sa-postgresql/db_name
+$ doguctl config -e sa-postgresql/db_name
+your-dogu-database-1234
+
+# schreibt Konfigurationswert in /config/<dogu>/my_key hinein
+$ doguctl config my_key 'my new value'
+
+# schreibt verschlüsselten Geheimnis /config/<dogu>/geheim/credential
+$ doguctl config -e geheim/credential '$up3r$3cre7'
+
+# löscht den Schlüssel /config/<dogu>/delete_me
+$ doguctl config --rm delete_me
+```
+
+#### doguctl validate
+
+Dieser Aufruf validiert Konfigurationswerte der Registry. Das Kommando `validate` liefert einen Exit Code == 0, wenn keines der überprüften Dogu-Konfigurationswerte einen Validierungsfehler aufweist. Ansonsten liefert es einen Exit Code == 1.
+
+```bash
+$ doguctl validate logging/root # validiert einzelnen Wert, Gutfall: Wert=WARN aus ERROR,WARN,INFO,DEBUG
+Key logging/root: OK
+$ echo $?
+0
+
+$ doguctl validate logging/root #  validiert einzelnen Wert, Fehlerfall: Wert=foo aus ERROR,WARN,INFO,DEBUG
+Key logging/root: INVALID
+at least one key was invalid
+$ echo $?
+1
+
+$ doguctl validate --all # validiert alle Werte mit Ausgabe
+...
+Key container_config/memory_limit: OK
+Key container_config/swap_limit: OK
+Key container_config/java_max_ram_percentage: OK
+Key container_config/java_min_ram_percentage: OK
+Key logging/root: INVALID
+Key caching/workdir/size: OK (No Validator)
+at least one key was invalid
+$ echo $?
+1
+
+$ doguctl validate --silent # validiert Werte ohne Ausgabe (kombinierbar mit --all), Gutfall
+$ echo $?
+0
+
+$ doguctl validate --silent # validiert Werte ohne Ausgabe (kombinierbar mit --all), Fehlerfall
+at least one key was invalid
+echo $?
+1
+```
+
+#### doguctl healthy
+
+Dieser Aufruf prüft, ob ein gegebenes Dogu betriebsbereit (healthy) ist. Das Kommando `healthy` liefert einen Exit Code == 0, wenn das betrachtete Dogu healthy ist. Ansonsten liefert es einen Exit Code == 1.
+
+```bash
+if ! doguctl healthy --wait --timeout 120 postgresql; then
+  echo "timeout reached by waiting of ldap to get healthy"
+  exit 1
+fi
+```
+
+#### doguctl state
+
+Dieser Aufruft liest und schreibt Dogu-Zustandswerte. Er wird in Kombination mit einem [HealthCheck](../core/compendium_de.md#healthchecks) verwendet.
+
+```bash
+$ doguctl state "installing" # schreibt den Wert in den State
+# Erledige Installation
+
+$ doguctl state # Liest den Wert aus dem State
+installing
+
+$ doguctl state "ready" # Setzt den State auf den Standard-Healthy-Wert
+# Starte nun die Anwendung
+```
+
+#### doguctl random
+
+Dieser Aufruf erzeugt Zufallsstrings, geeignet um Passwörter oder sonstige Zugangsdaten zu erzeugen.
+
+```bash
+$ doguctl random # erzeugt Zufallsstring mit der Länge 16
+9HoF4nYmlLYtf6Ju
+$ doguctl random  -l 5 # erzeugt Zufallsstring mit gegebener Länge
+W6Wmj
+```
```

Comment:
is this cryptographically secure?

Reply:
From a theoretical stance: not yet. For an cryptgraphically secure value the go library `crypto` must be used. 

From a practical stance: Safe enough given the unpredictability of a container runtime latency.